tests/printing/infix-operations.check

```diff
@@ -0,0 +1,35 @@
+[[syntax trees at end of                     typer]] // tests/printing/infix-operations.scala
+package <empty> {
+  class C(a: Int) extends Object() {
+    private[this] val a: Int
+    def foo(b: Int): C = this
+    def +(b: Int): C = this
+  }
+  final lazy module val infix-operations$package: infix-operations$package =
+    new infix-operations$package()
+  final module class infix-operations$package() extends Object() {
+    this: infix-operations$package.type =>
+    @main def main: Unit =
+      {
+        val v1: Int = 1 + 2 + 3
+        val v2: Int = 1 + (2 + 3)
+        val v3: Int = 1 + 2 * 3
+        val v4: Int = (1 + 2) * 3
+        val v5: Int = 1 + 2:Int
```

Comment:
aren't the parentheses important to preserve here?

Reply:
No, because `+` has a higher precedence than `:`, so `1 + 2 : Int` is the same as `(1 + 2) : Int`:

```
ValDef(v5, TypeTree(),
  Typed(Apply(Select(Literal(1), +), List(Literal(2))), Ident(Int))),
ValDef(v6, TypeTree(),
  Typed(Apply(Select(Literal(1), +), List(Literal(2))), Ident(Int)))
```

However it definitely seems confusing to me as well, so maybe we could print `Typed` with a higher precedence even if it's not required for correctness. I pushed a commit that does that.